subiquity/common/types.py

```diff
@@ -244,20 +244,29 @@ class Bootloader(enum.Enum):
 
 @attr.s(auto_attribs=True)
 class Partition:
-    size: int
-    number: int
-    annotations: List[str]
+    size: Optional[int] = None
+    number: Optional[int] = None
+    preserve: Optional[bool] = None
+    wipe: Optional[str] = None
+    annotations: Optional[List[str]] = []
```

Comment:
I guess this is OK in this case but for mutable defaults something like

```
    annotations: List[str] = attr.ib(default=attr.Factory(list))
```

is better. (Also I don't think the `Optional` is appropriate here?)